### PR TITLE
AccountPayment & Helper Interfaces

### PR DESCRIPTION
**Pull request by rhyswilliamsza**
Added interfaces for handling Amount & PaymentMethod models. Also added model for AccountPayment.

### CODE REVIEW

File: src/main/java/io/electrum/vas/model/AccountPayment.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+   }
+
+   /**
+    * AccountId to which this payment will be made.
+    *
+    * @return accountId
+    **/
+   @ApiModelProperty(required = true, value = "AccountId to which this payment will be made.")
+   @JsonProperty("accountId")
+   @NotNull
```
Comment (eon) [inline src/main/java/io/electrum/vas/model/AccountPayment.java:29]: think you might have to remove this to make it optional

Comment (rhyswilliamsza) [reply src/main/java/io/electrum/vas/model/AccountPayment.java]: Coolios, all done.

**Comment by TimElectrum** [comment]
I see that we're supporting account-to-account payments but we only indicate the "to" account, not the "from" account. That seems a little one sided. Should we not add support for the "from" account too?

**Comment by rhyswilliamsza** [comment]
> I see that we're supporting account-to-account payments but we only indicate the "to" account, not the "from" account. That seems a little one sided. Should we not add support for the "from" account too?

Makes sense. With the design as laid out in the current spec, we ironically assume the 'to' to be the merchant itself (thus, in this case, we're actually missing a 'from'). I will add it in for reusability regardless.

**Comment by TimElectrum** [comment]
> > I see that we're supporting account-to-account payments but we only indicate the "to" account, not the "from" account. That seems a little one sided. Should we not add support for the "from" account too?
> 
> Makes sense. With the design as laid out in the current spec, we ironically assume the 'to' to be the merchant itself (thus, in this case, we're actually missing a 'from'). I will add it in for reusability regardless.

I think it's fine for one of them to be missing in a specific implementation if there's an implicit understanding e.g. when this comes from a retailer, the retailer is _not_ listed as the "to" account because all customers are paying to the retailer.

But there should definitely still be fields for both.

**Comment by rhyswilliamsza** [comment]
> But there should definitely still be fields for both.

Totally agreed!

**Pull request merged**